tests/test_jobs.py

```diff
@@ -724,3 +720,346 @@ def test_transactions(self):
+
+        id6 = identities[5]
+        self.assertEqual(id6, self.jsm3)
+
+        id7 = identities[6]
+        self.assertEqual(id7, self.js_alt2)
+
+    def test_unify_empty_target(self):
+        """Check if unify is applied for the given individuals against the whole registry"""
+
+        ctx = SortingHatContext(self.user)
+
+        # Test
+        expected = {
+            'results': [
+                self.jsmith.uuid
+            ],
+            'errors': []
+        }
+
+        source_uuids = [self.john_smith.uuid]
+        target_uuids = None
+        criteria = ['email', 'name']
+
+        job = unify.delay(ctx,
+                          source_uuids,
+                          target_uuids,
+                          criteria)
+
+        result = job.result
+
+        self.assertDictEqual(result, expected)
+
+        # Checking if the identities have been merged
+        individual = Individual.objects.get(mk=self.jsmith.uuid)
+        identities = individual.identities.all()
+        self.assertEqual(len(identities), 6)
+
+        id1 = identities[0]
+        self.assertEqual(id1, self.jsm2)
+
+        id2 = identities[1]
+        self.assertEqual(id2, self.jsmith)
+
+        id3 = identities[2]
+        self.assertEqual(id3, self.jsm3)
+
+        id4 = identities[3]
+        self.assertEqual(id4, self.john_smith)
+
+        id5 = identities[4]
+        self.assertEqual(id5, self.js2)
+
+        id6 = identities[5]
+        self.assertEqual(id6, self.js3)
+
+    def test_no_matches_found(self):
+        """Check whether it returns no results when there is no matches for the input identity"""
+
+        ctx = SortingHatContext(self.user)
+
+        # Test
+        expected = {
+            'results': [],
+            'errors': []
+        }
+
+        source_uuids = [self.john_smith.uuid]
+        target_uuids = [self.jrae.uuid]
+        criteria = ['email', 'name']
+
+        job = unify.delay(ctx,
+                          source_uuids,
+                          target_uuids,
+                          criteria)
+
+        result = job.result
+
+        self.assertDictEqual(result, expected)
+
+    @unittest.mock.patch('sortinghat.core.api.find_individual_by_uuid')
+    def test_not_found_uuid_error(self, mock_find_indv):
+        """Check if the unify process returns no results when an individual is not found"""
+
+        exc = NotFoundError(entity='1234567890abcdefg')
+        mock_find_indv.side_effect = exc
+
+        ctx = SortingHatContext(self.user)
+
+        # Test
+        expected = {
+            'results': [],
+            'errors': []
+        }
+
+        source_uuids = ['1234567890abcdefg']
+        target_uuids = [self.john_smith.uuid]
+        criteria = ['email', 'name']
+
+        job = unify.delay(ctx,
+                          source_uuids,
+                          target_uuids,
+                          criteria)
+        result = job.result
+
+        self.assertDictEqual(result, expected)
+
```

Comment:
I was not sure about what to to with this test. It was meant to test what happens when an input individual is not found on the registry. However, when this happens it causes an error in the recommendation system and that is not captured here. If any errors were returned here, they would have been related with the merging process... WDYT @sduenas ?

Reply:
Maybe the method should return the errors produced during the recomendation? Does it make sense to you?

Reply:
I agree, but I couldn't find a simple way to include these errors in the result without modifying the recommendation engine. Do you think we should modify the engine to support it?